lib/player.rb

```diff
@@ -0,0 +1,93 @@
+require_relative "scoring"
+require_relative "tilebag"
+
+module Scrabble
+  class Player
+    attr_reader :name, :words_played, :score, :tiles
+
+    def initialize(name, restrict_tiles=false)
+      @name = name
+      @words_played = []
+      @score = 0
+      @tiles = []
+      @restrict_tiles = restrict_tiles
+    end
+
+    def plays
+      return @words_played
+    end
+
+    def play(word)
+      if check_word(word)
+        word_score = Scoring.score(word)
+        begin
+          @score += word_score
+          @words_played << word.upcase
+        rescue TypeError
+          puts "A valid word was not entered. You have lost your turn."
+          word_score = 0
+        end
+        word.to_s.upcase.chars.each do |letter|
+          i = @tiles.find_index(letter)
+          @tiles.delete_at(i || @tiles.length)
+        end
+        if won?
```

Comment:
I would check to see if you won **before** deleting tiles from the player's hand.